src/shared/containers/GraphContainer.tsx

```diff
@@ -120,7 +120,13 @@ const GraphContainer: React.FunctionComponent<{
     resource._self
   );
 
-  const [selectedResource, setSelectedResource] = React.useState<string>('');
+  const [{ selectedResourceId, isSelectedExternal }, setSelectedResource] = React.useState<{
```

Comment:
suggestion: This can be it own type.  This won't add anything to the functionality, but could be aesthetically pleasing.